com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StageAppStep.java

```diff
@@ -26,8 +27,11 @@
 
     @Override
     protected StepPhase executeAsyncStep(ExecutionWrapper execution) {
-        CloudApplication app = StepsUtil.getApp(execution.getContext());
-        ApplicationStager applicationStager = new ApplicationStager(execution.getControllerClient());
+        String appName = StepsUtil.getApp(execution.getContext())
+                                  .getName();
+        CloudControllerClient client = execution.getControllerClient();
+        CloudApplication app = client.getApplication(appName);
```

Comment:
Can we avoid the fetch of this app?

Reply:
We need the GUID...
It is in the ApplicationStager app.getMetadata().getGuid()